src/wasm/wasm-ir-builder.cpp

```diff
@@ -207,18 +207,33 @@ void IRBuilder::dump() {
 
   for (auto& scope : scopeStack) {
     std::cerr << "  scope ";
-    if (std::get_if<ScopeCtx::NoScope>(&scope.scope)) {
+    if (scope.isNone()) {
       std::cerr << "none";
-    } else if (auto* f = std::get_if<ScopeCtx::FuncScope>(&scope.scope)) {
-      std::cerr << "func " << f->func->name;
-    } else if (std::get_if<ScopeCtx::BlockScope>(&scope.scope)) {
+    } else if (auto* f = scope.getFunction()) {
+      std::cerr << "func " << f->name;
+    } else if (scope.getBlock()) {
       std::cerr << "block";
-    } else if (std::get_if<ScopeCtx::IfScope>(&scope.scope)) {
+    } else if (scope.getIf()) {
       std::cerr << "if";
-    } else if (std::get_if<ScopeCtx::ElseScope>(&scope.scope)) {
+    } else if (scope.getElse()) {
       std::cerr << "else";
-    } else if (std::get_if<ScopeCtx::LoopScope>(&scope.scope)) {
+    } else if (scope.getLoop()) {
       std::cerr << "loop";
+    } else if (auto* tryy = scope.getTry()) {
+      std::cerr << "try";
+      if (tryy->name) {
+        std::cerr << " " << tryy->name;
+      }
+    } else if (auto* tryy = scope.getCatch()) {
+      std::cerr << "catch";
+      if (tryy->name) {
+        std::cerr << " " << tryy->name;
+      }
+    } else if (auto* tryy = scope.getCatchAll()) {
+      std::cerr << "catch_all";
+      if (tryy->name) {
+        std::cerr << " " << tryy->name;
+      }
```

Comment:
`block` and `loop` names will be printed down below when we print the `OriginalLabel`. In contrast, `Try` names aren't considered `OriginalLabel`s because they are treated specially and used only for delegates, so if we want to print `Try` names we have to go out of our way to print them here.

Reply:
You said `originalLabel is the name as it appears in the text input.` above. Then a `try` doesn't have an `originalLabel` even if the input has one? I guess I'm confused because I don't understand the purpose of `originalLabel`.

Reply:
If I have this text input:

```
try $myTry
 try
 delegate $myTry
catch $tag
end
```

Then the `originalLabel` for that try is `myTry` and the `name` field in the `Try` IR node will be empty until we reach the `delegate`, at which point we will set it to `myTry` as well. Then when we finish the catch scope we will set the `name` to `__delegate__myTry` to differentiate it from the wrapper `block $myTry` that we use the `originalName` for.